lib/githubbot.ts

```diff
@@ -0,0 +1,403 @@
+/*
+Copyright 2016 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as ProcBot from './procbot';
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as fs from 'fs';
+import * as jwt from 'jsonwebtoken';
+import { AlertLevel } from './procbot';
+
+// The GithubAPI Promise implies built-in ES6 Promises. Therefore when assigning
+// the Promise to use as Bluebird, VS Code reports an error. Need a good
+// fix for this that doesn't require type declarations?
+const GithubApi = require('github');
+const request: any = Promise.promisifyAll(require('request'));
+
+// Github Events --------------------------------------------------------------
+// These provide the current bare minimum definitions for child Procbots working with them.
+export interface PullRequestEvent {
+    action: string,
+    pull_request: {
+        number: number,
+        head: {
+            repo: {
+                name: string;
+                owner: {
+                    login: string;
+                }
+            },
+            sha: string
+        }
+    }
+};
+
+export interface PullRequestReviewEvent {
+    action: string,
+    pull_request: {
+        number: number,
+        head: {
+            repo: {
+                name: string;
+                owner: {
+                    login: string;
+                }
+            },
+            sha: string
+        }
+    }
+};
+
+
+// Github API -----------------------------------------------------------------
+// These provide the current bare minimum definitions for child Procbots working with them.
+export interface CommitFile {
+    filename: string,
+};
+
+export interface Commit {
+    commit: {
+        committer: {
+            name: string
+        }
```

Comment:
Perhaps email as well? The procbot may need to alert the committer in some way that action is required.